reconcile/utils/terrascript_client.py

```diff
@@ -3892,12 +3917,72 @@ def populate_tf_resource_asg(self, resource, namespace_info):
                 {
                     "resource_type": "volume",
                     "tags": tags
-                }
-            ]
+                }]
         }
 
-        region = common_values.get('region') or \
-            self.default_regions.get(account)
+        image = common_values.get('image')
+        image_id = image.get('id')
+        if not image_id:
+            accounts = [a for a in self.filtered_accounts
+                        if a['name'] == account]
+            awsapi = AWSApi(1, accounts,
+                            settings=self.settings,
+                            init_users=False)
+            owner = image.get('owner')
+            if owner in self.uids.keys():
+                owner = self.uids.get(owner)
+
+            url = image.get('url')
+            ref = image.get('ref')
+            pattern = r'^[0-9a-f]{40}$'
+            upstream = image.get('upstream')
+            # get commit_sha from ref
+            if re.search(pattern, ref):
+                commit_sha = ref
+            # get commit_sha from upstream jenkins job
+            elif upstream:
+                instance_name = upstream['instance']['name']
+                token = upstream['instance']['token']
+                if instance_name not in self.jenkins_map:
+                    with self.jenkins_lock:
+                        if instance_name not in self.jenkins_map:
+                            self.jenkins_map[instance_name] = \
+                                JenkinsApi(token, ssl_verify=False,
+                                           settings=self.settings)
+                jenkins_instance = self.jenkins_map[instance_name]
+                job_name = upstream['name']
+                commit_sha = \
+                    jenkins_instance.get_last_success_build_commit(job_name)
+            # get commit_sha from branch
+            elif 'github' in url:
+                if not self.github:
+                    with self.github_lock:
+                        if not self.github:
+                            self.github = self.init_github()
+                repo_name = url.rstrip("/").replace('https://github.com/', '')
+                repo = self.github.get_repo(repo_name)
+                commit = repo.get_commit(sha=ref)
+                commit_sha = commit.sha
+            elif 'gitlab' in url:
+                if not self.gitlab:
+                    with self.gitlab_lock:
+                        if not self.gitlab:
+                            self.gitlab = self.init_gitlab()
+                project = self.gitlab.get_project(url)
+                commits = project.commits.list(ref_name=ref)
+                commit_sha = commits[0].id
+            # Get the most recent AMI named end with commit sha
+            name_regex = f".*{commit_sha}$"
+            image_ids = awsapi.get_image_ids(account, region, [owner],
```

Comment:
According to the way you fetch the owner, this can result in a call to `get_image_ids(account, region, [None], ...)`. Is this a valid call? What do you expect in such a case?

Reply:
if owner is none, aws will return all ami (public and private)